docker-storage-setup.sh

```diff
@@ -136,12 +136,14 @@ should_enable_deferred_deletion() {
 platform_supports_deferred_deletion() {
         local deferred_deletion_supported=1
         trap cleanup_pipes EXIT
-        mkfifo $PIPE1
-        mkfifo $PIPE2
-        mount -o bind $TEMPDIR $TEMPDIR
+
         if [ ! -x "/usr/lib/docker-storage-setup/dss-child-read-write" ];then
            return 1
         fi
+
+        mkfifo $PIPE1
+        mkfifo $PIPE2
+        mount -o bind $TEMPDIR $TEMPDIR
         unshare -m /usr/lib/docker-storage-setup/dss-child-read-write $PIPE1 $PIPE2 &
```

Comment:
That's a good point. We should not have to do mounting in host mount namespace. We should be able to create new mount namespace, do the mount and then signal to parent that try to remove directory from host mount namespace. 

Only extra overhead will be that we will have to pass the path to the tmp directory to child process. That should be fine. 

Will try it.